ml-proto/spec/int.ml

```diff
@@ -208,7 +208,19 @@ struct
   let ge_s x y = x >= y
   let ge_u x y = cmp_u x (>=) y
 
-  let of_string = Rep.of_string
+  (* to_string supports unsigned decimals *)
+  let of_string x =
+    let unsigned_decimal x =
+      let l = String.length x in
+      Rep.add
+        (Rep.mul
+          (Rep.of_string (String.sub x 0 (l - 1)))
+          (Rep.of_int 10))
+        (Rep.of_string (String.sub x (l - 1) 1))
+    in
+    try Rep.of_string x with
+      Failure _ -> unsigned_decimal x;;
```

Comment:
I think I'll prohibit negative numbers in unsigned_decimal. But I'm encountering some strange stuff: lexer.mll allows '+' in front of numbers, OCaml doesn't (atleast Rep.of_string doesn't), OCaml allows '_' within numbers, lexer.mll doesn't. (This would brake my current design.) Am I missing something, or is this some not intended behavior?